src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import usersFromServer from '../../api/users';
+import { getUserById } from '../../services/user';
+
+type Props = {
+  onSubmit: (todo: Todo) => void;
+};
+
+export const TodoForm: React.FC<Props> = ({ onSubmit }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onSubmit({
+      id: 0,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
+    });
+
+    setTitle('');
+    setUserId(0);
+
+    setHasTitleError(false);
+    setHasUserIdError(false);
+  };
+
+  return (
+    <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
+      <div className="field">
+        <input
+          type="text"
+          data-cy="titleInput"
+          placeholder="Enter a title"
+          value={title}
+          onChange={handleTitleChange}
+        />
+        {hasTitleError && <span className="error">Please enter a title</span>}
+      </div>
+
+      <div className="field">
+        <select
+          data-cy="userSelect"
+          value={userId}
+          onChange={handleUserIdChange}
+        >
+          <option value="0">Choose a user</option>
```

Comment:
The default option value is set to `0`, which is used to indicate that no user is selected. Ensure that the `usersFromServer` array is correctly populated and does not contain any invalid user IDs. This is important for maintaining data integrity and preventing errors during form submission.